src/main/java/com/ebivariation/contigalias/controller/AdminController.java

```diff
@@ -0,0 +1,53 @@
+package com.ebivariation.contigalias.controller;
+
+import com.ebivariation.contigalias.entities.AssemblyEntity;
+import com.ebivariation.contigalias.service.AssemblyService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+@RequestMapping("contig-alias-admin")
+@RestController
+public class AdminController {
+
+    private final AssemblyService service;
+
+    public AdminController(AssemblyService service) {
+        this.service = service;
+    }
+
+    @GetMapping(value = "assemblies/{accession}")
+    public Optional<AssemblyEntity> getAssemblyOrFetchByAccession(@PathVariable String accession) throws IOException {
```

Comment:
as a reminder to myself: check if Optional.empty() ends up being a http-404. If you try to getOrFetch something, and it's not present in any place, a 404 is probably the correct response.

Reply:
![check404](https://user-images.githubusercontent.com/31047659/83243820-149d5400-a1bc-11ea-8659-b85dab320fc2.png)
It is not a 404 but a 500

Reply:
ok then please change the controller to return 404 if empty.

Reply:
and add a test in ControllerIntegrationTest to check the return code.